include/hpp/constraints/convex-shape-contact.hh

```diff
@@ -156,6 +156,33 @@ namespace hpp {
 
       /// Display object in a stream
       std::ostream& print (std::ostream& o) const;
+
+      /// Currently only supports the case when all the joints for floors are the same
+      /// and all the joints for objects involved are the same
+      std::pair<JointConstPtr_t, JointConstPtr_t> jointsInvolved
+          (DeviceConstPtr_t robot) const {
```

Comment:
Why do you pass the robot as argument ?

Reply:
For `hpp::constraints::explicit_::ImplicitFunction` to detect locked joint I check that the input size of the function is 0, and the output range only involve exactly 1 joint (in which case this joint is locked with respect to the world joint). In order to retrieve the joint from the config rank, I need a pointer to the robot. I was also very reluctant to pass this as argument in the API since it only applies for that 1 function and does not apply for any other inheriting `DifferentiableFunction`...